src/main/java/org/skriptlang/skript/lang/experiment/ExperimentData.java

```diff
@@ -0,0 +1,153 @@
+package org.skriptlang.skript.lang.experiment;
+
+import ch.njol.skript.Skript;
+import ch.njol.util.StringUtils;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Container for holding {@link Experiment}s that must be enabled or disabled to use.
+ */
+public class ExperimentData {
+
+	private final Set<Experiment> required = new HashSet<>();
+	private final Set<Experiment> disallowed = new HashSet<>();
+	private @Nullable String errorMessage = null;
+
+	public ExperimentData() {}
+
+	/**
+	 * Set the {@link Experiment}s that must be enabled in order to use.
+	 * @param required The {@link Experiment}s
+	 * @return This {@link ExperimentData}.
+	 */
+	public ExperimentData required(Experiment... required) {
+		this.required.addAll(Arrays.stream(required).toList());
+		return this;
+	}
+
+	/**
+	 * Set the {@link Experiment}s that must be disabled in order to use.
+	 * @param disallowed The {@link Experiment}s
+	 * @return This {@link ExperimentData}.
+	 */
+	public ExperimentData disallowed(Experiment... disallowed) {
+		this.disallowed.addAll(Arrays.stream(disallowed).toList());
```

Comment:
this is now adding, not setting. Either it should set or this method's name/docs should be updated.